aip/0124.md

```diff
@@ -0,0 +1,116 @@
+---
+aip:
+  id: 124
+  state: reviewing
+  created: 2020-03-20
+permalink: /124
+redirect_from:
+  - /0124
+---
+
+# Resource association
+
+APIs sometimes have resource hierarchies that can not be cleanly expressed in
+the usual tree structure. For example, a resource may have a one-to-many
+relationship with two other resource types instead of just one. Alternatively,
+a resource may have a many-to-many relationship with another resource type.
+
+## Guidance
+
+A resource **must** have at most one canonical parent, and `List` requests
+**must not** require two distinct "parents" to work.
+
+### Multiple one-to-many associations
+
+If a resource has a one-to-many relationship with multiple resource types, it
+**must** choose at most one of them to be the canonical parent. The resource
+**may** be associated with other resources through other fields on the
+resource.
+
+```proto
+message Book {
+  // The resource name pattern for Book indicates that Publisher is the
+  // canonical parent.
+  option (google.api.resource) = {
+    type: "library.googleapis.com/Book"
+    pattern: "publishers/{publisher}/books/{book}"
+  };
+
+  // The resource name for the book.
+  string name = 1;
+
+  // The resource name for the book's author.
+  string author = 2 [(google.api.resource_reference) = {
```

Comment:
Possibly use the shorthand version of (google.api.resource_reference).type: "..."? (Ditto later annotations.)

Reply:
That goes beyond 80 characters.